autorest/adal/token.go

```diff
@@ -78,6 +84,17 @@ type Token struct {
 	Type     string `json:"token_type"`
 }
 
+// IsZero returns true if the token object is zero-initialized.
+func (t Token) IsZero() bool {
+	return len(t.AccessToken) == 0 &&
```

Comment:
For curiosity, why not just compare to an empty struct?

Reply:
That would work, however I think it feels a bit unnatural, i.e. I'd have to create some temporary Token first then compare the two.

Reply:
I'm kinda with @mcardosos on this one, implementing like this: https://play.golang.org/p/bH2OYqg6yv
would be durable to future elements being added to the Token struct, and would be a little easier to read IMHO.

Reply:
How about we make the guts of IsZero() use this comparison then?